core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -578,21 +581,29 @@ class LogManager(logDirs: Seq[File],
   }
 
   /**
-   * Make a checkpoint for all logs in provided directory.
-   */
+    * Make a checkpoint for all logs in provided directory and clean older snapshots for provided logs.
```

Comment:
Make a checkpoint for all logs in provided directory =>  Write the recovery checkpoint file for all logs in provided directory